### PR TITLE
[Unify Tensors PR #2] Replaced pten::LoD with paddle::framework::LoD

### PR DESCRIPTION
**Pull request by jim19930609**
### PR types
New features

### PR changes
Others

### Describe
The original paddle::framework::LoD is designed way more complex than the current pten::LoD. Therefore we temporarily rollbacked pten::LoD to paddle::framework::LoD, so as to facilitate the progress of Tensor unification.

### CODE REVIEW

File: paddle/fluid/framework/mixed_vector.h [hunk 1]
```diff
@@ -261,6 +261,7 @@ class Vector {
     }
 
     void CopyCPUDataToCUDA(const platform::Place &place) const {
+#if defined(PADDLE_WITH_CUDA) || defined(PADDLE_WITH_HIP)
```
Comment (Shixiaowei02) [inline paddle/fluid/framework/mixed_vector.h]: 建议 #elif 分支抛异常

File: paddle/fluid/framework/mixed_vector.h [hunk 2]
```diff
@@ -213,6 +211,7 @@ class Vector {
     };
 
     void CopyToCPU() const {
+#if defined(PADDLE_WITH_CUDA) || defined(PADDLE_WITH_HIP)
```
Comment (Shixiaowei02) [inline paddle/fluid/framework/mixed_vector.h]: 建议这个函数的实现放到 .cc

File: paddle/fluid/framework/mixed_vector.h (hunk 1 above)
Comment (Shixiaowei02) [inline paddle/fluid/framework/mixed_vector.h]: 建议带宏定义的相关实现放入 .cc

File: paddle/fluid/framework/mixed_vector.h [hunk 3]
```diff
@@ -23,17 +23,15 @@ limitations under the License. */
 
 #include "glog/logging.h"
 #include "paddle/fluid/framework/details/cow_ptr.h"
-#include "paddle/fluid/framework/tensor.h"
-#include "paddle/fluid/framework/tensor_util.h"
 #include "paddle/fluid/memory/malloc.h"
 #include "paddle/fluid/memory/memcpy.h"
+#include "paddle/fluid/platform/device_context.h"
```
Comment (Shixiaowei02) [inline paddle/fluid/framework/mixed_vector.h]: LoD 需要解耦头文件

File: paddle/fluid/framework/mixed_vector.h (hunk 1 above)
Comment (jim19930609) [reply paddle/fluid/framework/mixed_vector.h]: Done

File: paddle/fluid/framework/mixed_vector.h (hunk 2 above)
Comment (jim19930609) [reply paddle/fluid/framework/mixed_vector.h]: Done

**Review by Shixiaowei02 — APPROVED** [review]

**Pull request merged**